### PR TITLE
Qute - make the set of escaped content-types configurable

### LINKED ISSUE #21731: Qute should allow configuring HTML escaping for "custom" content-types/variants
### Description

Currently hard-coded here for xml and html: https://github.com/quarkusio/quarkus/blob/2.5.0.Final/independent-projects/qute/core/src/main/java/io/quarkus/qute/HtmlEscaper.java#L42

When you have e.g. xhml, you'll probably configure something like:
```
quarkus:
  qute:
    suffixes:
      - xhtml
    content-types:
      xhtml: application/xhtml+xml
```
but you'd still want escaping for those xhtml templates, which doesn't work, even not with a selected variant (at least for `@CheckedTemplates`; see #21730).

### Implementation ideas

```
quarkus:
  qute:
    html-escape-content-types:
      - xhtml
```
(or a better name)

Maybe it's better to create a new `content-types` config object that takes:
- file ending
- content-type
- locale (optional)
- encoding (optional)
- escaping yes/no (optional)

Otherwise the config would become really verbose and not so DRY, IMHO.

Also, I think `Variant` should be extended to toggle escaping.

Related:
- #21730
- #21405
- https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/Qute.20and.20XML.20templates.20.28char.20escaping.20issues.29

### PR DESCRIPTION
**Pull request by mkouba**
- introduce the quarkus.qute.escape-content-types configuration property
- resolves #21731

### CODE REVIEW

File: extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/EscapingTest.java [hunk 1]
```diff
@@ -26,11 +26,13 @@
...
                             "templates/foo.html")
                     .addAsResource(new StringAsset("{item} {item.raw}"),
-                            "templates/item.html")
+                            "templates/item.xhtml")
                     .addAsResource(new StringAsset("{text} {other} {text.raw} {text.safe} {item.foo}"),
                             "templates/bar.txt")
                     .addAsResource(new StringAsset("{@java.lang.String text}{text} {text.raw} {text.safe}"),
-                            "templates/validation.html"));
+                            "templates/validation.html"))
+            .overrideConfigKey("quarkus.qute.content-types.xhtml", "application/xhtml+xml")
```
Comment (gsmet) [inline extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/EscapingTest.java:34]: Didn't you add this one to the default value? If so, are you really testing your config?

Comment (mkouba) [reply extensions/qute/deployment/src/test/java/io/quarkus/qute/deployment/EscapingTest.java:34]: > Didn't you add this one to the default value?

Not really. This one maps `xhtml` files to `application/xhtml+xml`. It's a different config property.

**Review by famod — APPROVED** [review]
LGTM, thanks!

**Comment by famod** [comment]
@mkouba should`.xhtml` be a part of the `suffixes` default value, given that `application/xhtml+xml` is now a part of the `content-types` default value?

**Comment by mkouba** [comment]
> @mkouba should`.xhtml` be a part of the `suffixes` default value, given that `application/xhtml+xml` is now a part of the `content-types` default value?

Hmm, IDK. `.xhtml` is probably not so commonly used these days and I'd like to keep this list short and concise. On the other hand, it would not harm anything. At runtime the list is only used to locate a template if no suffix is used, e.g. if you attempt to locate a template `foo` then we try `foo.txt` etc. as well. If we add `.xhtml` then we should also add `.xml` and `.qute.xhtml` and `.qute.xml`... which is -0.5 from me.

**Comment by famod** [comment]
Alright! I'll merge this as that tck failure is not related.

**Pull request merged**